libs/api/ebpf_api.cpp

```diff
@@ -1475,6 +1481,169 @@ _get_next_map_to_create(std::vector<ebpf_map_t*>& maps)
+    _Outptr_result_maybenull_z_ const char** log_buffer)
+{
+    ebpf_result_t result = EBPF_SUCCESS;
+    std::vector<original_fd_handle_map_t> handle_map;
+    uint32_t error_message_size = 0;
+
+    *log_buffer = nullptr;
+
+    for (auto& program : object->programs) {
+        result = _create_program(
+            program->program_type, object->object_name, program->section_name, program->program_name, &program->handle);
+        if (result != EBPF_SUCCESS) {
+            break;
+        }
+
+        program->fd = _create_file_descriptor_for_handle(program->handle);
+
+        // Populate load_info.
+        ebpf_program_load_info load_info = {0};
+        load_info.object_name = const_cast<char*>(object->object_name);
+        load_info.section_name = const_cast<char*>(program->section_name);
+        load_info.program_name = const_cast<char*>(program->program_name);
+        load_info.program_type = program->program_type;
+        load_info.program_handle = reinterpret_cast<file_handle_t>(program->handle);
+        load_info.execution_type = execution_type;
+        load_info.byte_code = program->byte_code;
+        load_info.byte_code_size = program->byte_code_size;
+        load_info.execution_context = execution_context_kernel_mode;
+        load_info.map_count = (uint32_t)object->maps.size();
+
+        if (load_info.map_count > 0) {
+            for (auto& map : object->maps) {
+                fd_t inner_map_original_fd = (map->inner_map) ? map->inner_map->original_fd : ebpf_fd_invalid;
+                handle_map.emplace_back(
+                    map->original_fd, inner_map_original_fd, reinterpret_cast<file_handle_t>(map->map_handle));
+            }
+
+            load_info.handle_map = handle_map.data();
+        }
+
+        result = ebpf_rpc_load_program(&load_info, log_buffer, &error_message_size);
+        if (result != EBPF_SUCCESS) {
+            break;
+        }
+    }
+
+    if (result == EBPF_SUCCESS) {
+        for (auto& program : object->programs) {
+            _ebpf_programs.insert(std::pair<ebpf_handle_t, ebpf_program_t*>(program->handle, program));
+        }
+    }
+    return result;
+}
+
+ebpf_result_t
+ebpf_object_load(
+    _Inout_ struct bpf_object* object,
+    ebpf_execution_type_t execution_type,
+    _Outptr_result_maybenull_z_ const char** log_buffer)
+{
+    if (!object)
+        return EBPF_INVALID_ARGUMENT;
+
+    if (object->loaded) {
+        return EBPF_INVALID_ARGUMENT;
+    }
+
+    ebpf_result_t result = EBPF_SUCCESS;
+    try {
+        result = _ebpf_object_create_maps(object);
+        if (result != EBPF_SUCCESS) {
+            goto Done;
+        }
+
+        result = _ebpf_object_load_programs(object, execution_type, log_buffer);
+    } catch (const std::bad_alloc&) {
+        result = EBPF_NO_MEMORY;
+        goto Done;
+    } catch (...) {
+        result = EBPF_FAILED;
+        goto Done;
+    }
+
+    object->loaded = true; /* doesn't matter if successfully loaded or not */
```

Comment:
Why does the comment mention that is does not matter if successfully loaded or not?

Reply:
This comment is identical to what is in https://github.com/libbpf/libbpf/blob/master/src/libbpf.c#L6953
The longer term intent is that our code will go away once we can just use the libbpf code directly.  So minimizing diffs is a good thing.